x/mongo/driver/topology/topology_options.go

```diff
@@ -68,8 +68,7 @@ func WithConnString(fn func(connstring.ConnString) connstring.ConnString) Option
 			c.serverOpts = append(c.serverOpts, WithServerAppName(func(string) string { return cs.AppName }))
 		}
 
-		switch cs.Connect {
-		case connstring.SingleConnect:
+		if cs.Connect == connstring.SingleConnect || (cs.DirectConnectionSet && cs.DirectConnection) {
```

Comment:
Can this be tested?  Perhaps it is, but if so it wasn't obvious to me where.

Reply:
Added tests for this.